### DISCUSSION TITLE
How to configure executable release builds e.g. only build static executable, add shell completion

### DISCUSSION THREAD

**Discussion by raehik** [srid/haskell-flake#56] [Q&A]
I'm new to Nix. I've had great success using haskell-flake to succintly define the build environment for some packages. I'm not sure if it's meant for it, but I'm also trying to use it to configure release builds for packages which build both a library and a standalone binary.

I see in nixpkgs that the [pandoc CLI](https://github.com/NixOS/nixpkgs/blob/688e31f07d2ea0410d2cc0c2768a52c73f3592a8/pkgs/development/tools/pandoc/default.nix) has lots of configuration to move completions and man pages around. How may I do that in haskell-flake? (I can't seem to use the same attributes, and I was unable to find docs or schemas descriptions/expected types.)

Flake for the main project I'm trying to use this with: https://github.com/raehik/bytepatch/blob/d2b7bfff3c0d3194ae22bc87ec9ca63fee6c811c/flake.nix

**Comment by srid** [comment]
You can do that in `overrides` attribute, no?

See how Emanote does it: https://github.com/srid/emanote/blob/584d52c4c7071e643ce738db9db0196f17071518/flake.nix#L71-L80

**Comment by roberth** [comment]
If a package defines both an executable and a library, a single override may not be sufficient. For this reason it'd make sense to have something that only applies to the final packages, between extraction from the haskell package set and insertion into the `perSystem.packages` option.
